receiver/podmanreceiver/receiver.go

```diff
@@ -57,11 +66,73 @@ func newReceiver(
 		set:           set,
 	}
 
-	scrp, err := scraperhelper.NewScraper(typeStr, recv.scrape, scraperhelper.WithStart(recv.start))
+	return recv, err
+}
+
+func (r *receiver) registerMetricsConsumer(mc consumer.Metrics, set component.ReceiverCreateSettings) error {
+	r.metricsConsumer = mc
+	scrp, err := scraperhelper.NewScraper(typeStr, r.scrape, scraperhelper.WithStart(r.start))
 	if err != nil {
-		return nil, err
+		return err
+	}
+	r.metricsComponent, err = scraperhelper.NewScraperControllerReceiver(&r.config.ScraperControllerSettings, set, mc, scraperhelper.AddScraper(scrp))
+	return err
+}
+
+func (r *receiver) registerLogsConsumer(lc consumer.Logs) {
+	r.logsConsumer = lc
+}
+
+func (r *receiver) Start(ctx context.Context, host component.Host) error {
+	// Check for logs pipeline
+	if r.logsConsumer == nil {
+		r.set.Logger.Warn("Logs receiver is not set")
+	} else {
+		eventBackoff := backoff.NewExponentialBackOff()
+		eventBackoff.InitialInterval = 2 * time.Second
```

Comment:
Should this all be configurable?

Reply:
We decided to go with non-configurable, considering the below scenarios:
- User keeps the configuration where receiver stops retrying after x hours and podman comes up after x+1 hours, user would have to start the collector again.
- Podman keeps restarting and we configure receiver to stop retrying after short period, user would have to keep starting the collector again and again.

Due to above reasons we kept it non-configurable.